charmarchive.go

```diff
@@ -81,7 +81,7 @@ func readCharmArchive(zopen zipOpener) (archive *CharmArchive, err error) {
 	// this charm is v1 or not.
 	reader, err = zipOpenFile(zipr, "manifest.yaml")
 	if _, ok := err.(*noCharmArchiveFile); ok {
-		b.manifest = NewManifest()
```

Comment:
Do we still need `NewManifest` if it's never non-empty?

Reply:
No idea, will check.